scripts/irods/lib.py

```diff
@@ -786,6 +786,19 @@ def make_arbitrary_file(f_name, f_size, buffer_size=32*1024*1024):
 
             bytes_written += to_write
 
+def create_ascii_printable_file(filename, filesize):
+    """Creates a file with pseudo-random ASCII printable characters.
+    
+    - Does NOT use OS entropy (no `os.urandom` or `secrets`).
+    - Uses `random` for deterministic pseudo-randomness.
+    """
+    import string
+    # This avoids control sequences.
+    safe_chars = string.digits + string.ascii_letters + string.punctuation + " "
+    with open(filename, "w") as f:
+        for _ in range(filesize):
+            f.write(random.choice(safe_chars))
```

Comment:
I think we could use `choices(...)` here instead.
This would remove the for loop.

See: https://docs.python.org/3/library/random.html#random.choices

Reply:
I think you may have to change `write(...)` to `writelines(...)` too if you decide to use `choices(...)`.

Reply:
Updated.